sdk/src/Services/DynamoDBv2/Custom/DocumentModel/JsonUtils.cs

```diff
@@ -320,71 +330,81 @@ internal static void WriteJson(DynamoDBEntry entry, JsonWriter writer, DynamoDBE
             {
                 var itemType = primitiveList.Type;
 
-                writer.WriteArrayStart();
+                writer.WriteStartArray();
                 foreach (var item in primitiveList.Entries)
                 {
                     var itemValue = item.Value;
                     WritePrimitive(writer, itemType, itemValue);
                 }
-                writer.WriteArrayEnd();
+                writer.WriteEndArray();
                 return;
             }
 
             var ddbList = entry as DynamoDBList;
             if (ddbList != null)
             {
-                writer.WriteArrayStart();
-                foreach(var item in ddbList.Entries)
+                writer.WriteStartArray();
+                foreach (var item in ddbList.Entries)
                 {
                     WriteJson(item, writer, conversion);
                 }
-                writer.WriteArrayEnd();
+                writer.WriteEndArray();
                 return;
             }
 
             var ddbBool = entry as DynamoDBBool;
             if (ddbBool != null)
             {
-                writer.Write(ddbBool.Value);
+                writer.WriteBooleanValue(ddbBool.Value);
                 return;
             }
 
             var ddbNull = entry as DynamoDBNull;
             if (ddbNull != null)
             {
-                writer.Write((string)null);
+                writer.WriteNullValue();
                 return;
             }
 
             throw new InvalidOperationException(string.Format(CultureInfo.InvariantCulture,
                 "Unable to convert entry of type {0} to JSON", entry.GetType().FullName));
         }
 
-        // Write the contents of a Primitive object as JSON data
-        private static void WritePrimitive(JsonWriter writer, DynamoDBEntryType type, object value)
+        /// <summary>
+        /// Write the contents of a Primitive object as JSON data
+        /// </summary>
+        private static void WritePrimitive(Utf8JsonWriter writer, DynamoDBEntryType type, object value)
         {
             var stringValue = value as string;
 
             switch (type)
             {
                 case DynamoDBEntryType.Numeric:
-                    writer.WriteRaw(stringValue);
-                    break;
+                    {
+#if NETCOREAPP3_1  // WriteRawValue was added in .NET 6, but we need to write out Number values without quotes
```

Comment:
Question: Does this mean for the .NET Framework target we'd try to use `WriteRawValue` too (which is only available in .NET 6)? Or is it fine because we have `System.Text.Json` as a dependency for `net472`?

Reply:
It's [available in 4.7.2](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonwriter.writerawvalue?view=net-8.0#system-text-json-utf8jsonwriter-writerawvalue(system-string-system-boolean). At the time 3.1 was the only target we needed this workaround for.